### PR TITLE
Add Base Class for WebSocket Unit testing

### PR DESCRIPTION
**Pull request by ManosMagnus**
Used to establish WebSockcet connection and send messages

### CODE REVIEW

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.io.IOException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.websocket.client.HandshakeConfig;
+import org.zaproxy.zap.extension.websocket.client.HttpHandshakeBuilder;
+import org.zaproxy.zap.extension.websocket.client.ServerConnectionEstablisher;
+import org.zaproxy.zap.testutils.WebSocketTestUtils;
+
+public abstract class WebSocketAddonTestUtils extends WebSocketTestUtils {
+
+    public WebSocketProxy setUpWebSocketProxy() {
```
Comment (kingthorin) [inline addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: Really minor details but `setup` is one word, so the U doesn't really need to be caps.

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ServerConnectionEstablisher establisher = new ServerConnectionEstablisher();
+        try {
+            HttpMessage handshakeRequest =
+                    new HttpMessage(
+                            HttpHandshakeBuilder.getHttpHandshakeRequestHeader(
+                                    super.getServertUrl()));
```
Comment (kingthorin) [inline addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: Extra `t` in `Servert`?

Comment (ManosMagnus) [reply addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: Right, I've noticed as well. Is it OK to tweak them in this commit?

Comment (kingthorin) [reply addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: I don't see any reason not to. You can do an amending commit or just a new push and squash it later.

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            return establisher.send(new HandshakeConfig(handshakeRequest, false, false));
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public WebSocketMessageDTO sendMessage(WebSocketProxy webSocketProxy, String message) {
```
Comment (thc202) [inline addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: `sendOutgoingMessage`?

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        WebSocketMessageDTO webSocketMessage = new WebSocketMessageDTO(webSocketProxy.getDTO());
+        webSocketMessage.payload = message;
+        webSocketMessage.opcode = WebSocketMessage.OPCODE_TEXT;
+        webSocketMessage.isOutgoing = true;
+
+        try {
+            return (webSocketProxy.send(webSocketMessage, WebSocketProxy.Initiator.MANUAL_REQUEST))
```
Comment (thc202) [inline addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: The extra parentheses could be removed.

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java (hunk 2 above)
Comment (ManosMagnus) [reply addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: OK, I've keep it in a separate commit for now. Thank u

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java (hunk 1 above)
Comment (ManosMagnus) [reply addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: Fixed

File: addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java (hunk 3 above)
Comment (ManosMagnus) [reply addOns/websocket/src/test/java/org/zaproxy/zap/extension/websocket/WebSocketAddonTestUtils.java]: fixed

**Comment by thc202** [comment]
Thanks!

**Pull request merged**